chia/server/node_discovery.py

```diff
@@ -490,47 +486,33 @@ async def _add_peers_common(
             await self.address_manager.add_to_new_table(peers_adjusted_timestamp, None, 0)
 
 
+@dataclass
 class FullNodePeers(FullNodeDiscovery):
-    self_advertise_task: Optional[asyncio.Task[None]] = None
-    address_relay_task: Optional[asyncio.Task[None]] = None
-
-    def __init__(
-        self,
-        server: ChiaServer,
-        target_outbound_count: int,
-        peers_file_path: Path,
-        introducer_info: dict[str, Any],
-        dns_servers: list[str],
-        peer_connect_interval: int,
-        selected_network: str,
-        default_port: Optional[int],
-        log: Logger,
-    ) -> None:
-        super().__init__(
-            server,
-            target_outbound_count,
-            peers_file_path,
-            introducer_info,
-            dns_servers,
-            peer_connect_interval,
-            selected_network,
-            default_port,
-            log,
-        )
-        self.relay_queue = asyncio.Queue()
-        self.neighbour_known_peers: dict[PeerInfo, set[str]] = {}
-        self.key = randbits(256)
+    self_advertise_task: Optional[asyncio.Task[None]] = field(default=None)
+    address_relay_task: Optional[asyncio.Task[None]] = field(default=None)
+    relay_queue: asyncio.Queue[tuple[TimestampedPeerInfo, int]] = field(default_factory=asyncio.Queue)
+    neighbour_known_peers: dict[PeerInfo, set[str]] = field(default_factory=dict)
+    key: int = field(default_factory=lambda: random.getrandbits(256))
 
-    async def start(self) -> None:
-        await self.initialize_address_manager()
-        self.self_advertise_task = create_referenced_task(self._periodically_self_advertise_and_clean_data())
-        self.address_relay_task = create_referenced_task(self._address_relay())
-        await self.start_tasks()
+    def __post_init__(self) -> None:
+        super().__post_init__()
```

Comment:
Is this strictly necessary?  I can see it would have some benefits around refactors where it would remind whoever touches it to consider the inheritance.  Just curious if you actually ran into any issues without this.

Reply:
I don't think it's strictly necessary, no